File: typescript/src/PlanetTS1/Model/Neutral/Buildings/GasCollector.ts


abstract class GasCollector extends Building {
    minCost : number
    gasCost : number
    nrgCost : number
    commands : Array<string>
    health : number

    constructor() {
        super()
        this.minCost = 0
        this.gasCost = 0
        this.nrgCost = 0
        this.commands = new Array<string>("Abort")
        this.health = 100
    }
}

File: dotnet/src/Model/Neutral/Buildings/GasCollector.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    public abstract class GasCollector : Building {
        public GasCollector() {
            this.minCost = 0;
            this.gasCost = 0;
            this.nrgCost = 0;
            this.commands = new List<String>(){"Abort"};
            this.health = 100;
        }
    }
}